programs/fileio.c

```diff
@@ -2558,12 +2569,23 @@ FIO_decompressMultipleFilenames(FIO_prefs_t* const prefs,
             EXM_THROW(72, "Write error : %s : cannot properly close output file",
                         strerror(errno));
     } else {
-        unsigned u;
+        unsigned int u = 0;
+        if (outMirroredRootDirName)
+            UTIL_mirrorSourceFilesDirectories(srcNamesTable, nbFiles, outMirroredRootDirName);
+
         for (u=0; u<nbFiles; u++) {   /* create dstFileName */
             const char* const srcFileName = srcNamesTable[u];
-            const char* const dstFileName = FIO_determineDstName(srcFileName, outDirName);
+            const char* dstFileName = NULL;
+            if (outMirroredRootDirName) {
+                char* outMirroredDirName = UTIL_createMirroredDestDirName(srcFileName, outMirroredRootDirName);
+                if (outMirroredDirName) {
+                    dstFileName = FIO_determineDstName(srcFileName, outMirroredDirName);
+                    free(outMirroredDirName);
+                }
```

Comment:
what happens `if (!outMirroredDirName)` ?

I believe it's reasonable to stop execution if such a case happen, as we probably don't want to continue execution if the system can't even allocate a bit of memory for a file name.

Reply:
here the 'error=1' after outMirroredRootDirName, I think it will cause the program return non-0 error status.

For malloc issues, it the program will call exit.

The user case of 'outMirroredRootDir == NULL" is actually because the input dir has "..", according to the `tar`, it continue to work with other inputs. Should we follow the `tar` behavior? 

But I think need add an error print here.

Reply:
Yes, an information message is a good idea.
Use level 2, as it's a normal warning, not a fatal error.